### DISCUSSION TITLE
Does osm2pgsql support postgresql 14?

### DISCUSSION THREAD

**Discussion by chenrui333** [osm2pgsql-dev/osm2pgsql#1575] [Q&A] [answered]
When [we are testing postgresql v14 upgrade PR](https://github.com/Homebrew/homebrew-core/pull/86220), we found some regression test build failure with osm2pgsql. Thus, raising a thread in here to double before moving forward with the debugging process. Thanks!

**Comment by joto** [comment]
I am not aware of anybody who has tried running it on PostgreSQL 14. What problems are you getting?

**Comment by stephankn** [comment]
To my understanding, this was an issue with a changed wording of the "connection failed" error message. So no problem with osm2pgsql:
https://github.com/Homebrew/homebrew-core/pull/86220/commits/137d3d4b258a944c4a21c9e2c329480122ff81e8

**Comment by rustprooflabs** [comment, marked as answer]
I upgraded my osm2pgsql setup from Pg13 to Pg14 a few days ago, no issues so far.

**Comment by nmaludy** [comment]
I upgraded from 13 -> 14 using pg_upgrade a d started running into Out Of Memory errors when rendering new tiles. I tried increasing every parameter I could in the config and never got it working. My gut told me it was a PostGIS issue, but have no proof.

Next time I'm going to try importing directly into a fresh 14 database rather than upgrading and see how it goes.

**Reply by pnorman** [reply]
Issues when rendering wouldn't be related to osm2pgsql, as osm2pgsql doesn't render tiles.

**Comment by SomeoneElseOSM** [comment]
Out of the box Ubuntu 22.04 uses:

    psql --version
    psql (PostgreSQL) 14.3 (Ubuntu 14.3-0ubuntu0.22.04.1)

I can't say that whether it needs more or less memory (or runs slower or faster), but a simple "soup to nuts" test as per https://switch2osm.org/serving-tiles/manually-building-a-tile-server-ubuntu-22-04-lts/ works OK.
